internal/provider/helpers.go

```diff
@@ -0,0 +1,84 @@
+package provider
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/gookit/color"
+)
+
+type tfOperation string
+
+const (
+	CreateOperation tfOperation = "CREATE"
+	ReadOperation   tfOperation = "READ"
+	UpdateOperation tfOperation = "UPDATE"
+	DeleteOperation tfOperation = "DELETE"
+)
+
+func formatSQLPreview(operation tfOperation, resourceName string, id string, commands []string) string {
+	var c color.Color
+	switch operation {
+	case CreateOperation:
+		c = color.HiGreen
+	case ReadOperation:
+		c = color.HiBlue
+	case UpdateOperation:
+		c = color.HiYellow
+	case DeleteOperation:
+		c = color.HiRed
+	}
+	var sb strings.Builder
+	sb.WriteString(c.Sprintf("\n[ %s %s %s ]", operation, resourceName, id))
+	for _, command := range commands {
+		sb.WriteString(c.Sprintf("\n  - %s", command))
+	}
+	sb.WriteString("\n")
+	return sb.String()
+}
+
+type sensitiveAttributes struct {
+	m map[string]bool
+}
+
+var (
+	sa   *sensitiveAttributes
+	lock = sync.Mutex{}
```

Comment:
+1 for init function and logic simplification

Reply:
i was just trying to get a demonstration of how this might work, I don't even know if we should be doing this or not. we should discuss with the tf provider framework team when we have a meeting